db/migrate/20191216135816_add_checked_by_reference_to_checks.rb

```diff
@@ -0,0 +1,11 @@
+class AddCheckedByReferenceToChecks < ActiveRecord::Migration[6.0]
+  def up
+    add_column :checks, :checked_by_id, :uuid, using: "checked_by::uuid"
```

Comment:
I'm pretty sure we could use `add_reference` to do both things in one go and use `change` instead to get the automagic migrate down.

Reply:
We could, but this would result in a field called `dfe_sign_in_user_id`, which doesn't really reflect the intention of the column.

Reply:
Does the `add_reference` API not allow the foreign key to be overridden?

Reply:
Had a look at the [API](https://apidock.com/rails/ActiveRecord/ConnectionAdapters/SchemaStatements/add_reference), I think this would do it:

```suggestion
  def change
    add_reference :checks, :checked_by, type: :uuid, foreign_key: {to_table: :dfe_sign_in_users}
  end
```

Reply:
Arrrrgh. That's got it. I was looking at the wrong version of the docs.